### PR TITLE
Add arrow and offsets props to the Popover

### PR DESCRIPTION
**Pull request by Klynger**
#### What does this PR do? \*

As the title says.

This is the config of all three `locale-switcher`s that are below the shelf (the header is the same as the first `locale-switcher`)
```diff
{
"popover-layout#locale-switcher": {
    "children": [
      "locale-switcher-list"
    ],
    "props": {
      "scrollBehavior": "close-on-scroll",
      "backdrop": "visible",
+     "arrow": true,
+     "offsets": {
+       "distance": 20,
+       "skidding": 30
+    }
    }
  },
  "popover-layout#locale-switcher2": {
    "children": [
      "locale-switcher-list"
    ],
    "props": {
      "scrollBehavior": "lock-page-scroll",
      "backdrop": "visible",
      "blockClass": "flyout",
+    "arrow": true,
+    "offsets": {
+      "distance": 5
+    }
    }
  },
  "popover-layout#locale-switcher3": {
    "children": [
      "locale-switcher-list"
    ],
    "props": {
      "placement": "left",
      "backdrop": "visible",
+    "arrow": true,
+    "offsets": {
+      "distance": 5
      }
    }
  }
}
```

#### How to test it? \*

1. [workspace](https://popoverswitcher--storecomponents.myvtex.com/)
2. open the `locale-switcher` from the header and the ones that are below the shelf

### CODE REVIEW

**Review by iaronaraujo — APPROVED** [review]

File: react/Popover.tsx [hunk 1]
```diff
@@ -13,10 +13,17 @@ import handleContainerStyle, { RestoreFn } from './modules/handleContainerStyle'
...
 
+interface Offsets {
+  distance: number
+  skidding: number
+}
+
 interface Props {
   role?: string
-  placement: Placement
+  arrow?: boolean
```
Comment (victorhmp) [inline react/Popover.tsx]: Any chance of this becoming an `Enum`? Just bringing this up, I can't think of a use case.
I would at least make this responsive, what do you think?

File: react/Popover.tsx [hunk 2]
```diff
@@ -34,27 +41,25 @@ function useTransition(option: TransitionComponentType) {
   }, [option])
 }
 
-const CSS_HANDLES = ['container']
+const CSS_HANDLES = ['container', 'arrow']
```
Comment (victorhmp) [inline react/Popover.tsx]:
```suggestion
const CSS_HANDLES = ['container', 'arrow'] as const
```

File: docs/README.md [hunk 3]
```diff
@@ -59,21 +59,24 @@ This repository provides blocks that can help you create a Dropdown, Select or a
 | `placement` | `Placement` | Describes the preferred placement of the popover. If there is no space to put the popover in the placement that you choose it will fit in a fallback position | `'bottom`' |
 | `scrollBehavior` | `ScrollBehavior` | How the popover should behave if the user tries to scroll the page | `'default'` |
 | `backdrop` | `BackdropOptions` | If it should display a backdrop overlay | `'none'` |
+| `arrow`  | `boolean` | If it should display an arrow pointing to the `popover-trigger` element | `false` |
```
Comment (victorhmp) [inline docs/README.md]: Maybe `arrow` is a too generic name. How about `showArrow`, or `popoverArrow`?

File: react/Trigger.tsx [hunk 4]
```diff
@@ -12,7 +12,7 @@ import {
   PopoverContextProvider,
 } from './PopoverContext'
 
-const CSS_HANDLES = ['triggerContainer', 'trigger']
+const CSS_HANDLES = ['trigger']
```
Comment (victorhmp) [inline react/Trigger.tsx]:
```suggestion
const CSS_HANDLES = ['trigger'] as const
```

File: react/components/Popper.tsx [hunk 5]
```diff
@@ -39,23 +46,32 @@ interface Props
...
+  arrowEl?: HTMLElement | null
   placement?: PopperPlacementType
 }
 
 function getAnchorEl(anchorEl: React.ReactNode) {
   return typeof anchorEl === 'function' ? anchorEl() : anchorEl
 }
 
-const CSS_HANDLES = ['popper']
+const CSS_HANDLES = ['popper', 'popperArrow']
```
Comment (victorhmp) [inline react/components/Popper.tsx]:
```suggestion
const CSS_HANDLES = ['popper', 'popperArrow'] as const
```

**Review by klzns — CHANGES_REQUESTED** [review]
It is creating a big horizontal scroll

image

File: react/components/Popper.tsx [hunk 6]
```diff
@@ -77,6 +93,51 @@ const Popper = forwardRef(function Popper(props: Props, ref) {
...
+        enabled: true,
+        phase: 'afterWrite',
+        fn: handlePopperUpdate,
+      },
+      {
+        name: 'preventOverflow',
+        options: {
+          mainAxis: true,
+          altAxis: false,
+          ...(window && { boundary: window }),
```
Comment (victorhmp) [inline react/components/Popper.tsx:118]: If `window` is `undefined`, what would be the desired behavior? I mean, would this work without the `boundary` property?

**Review by victorhmp — CHANGES_REQUESTED** [review]

File: react/Popover.tsx (hunk 2 above)
Comment (Klynger) [reply react/Popover.tsx]: That doesn't change anything since we don't type the `useCssHandles` right

File: react/components/Popper.tsx (hunk 6 above)
Comment (Klynger) [reply react/components/Popper.tsx:118]: Nice catch, this doesn't change anything

File: docs/README.md [hunk 7]
```diff
@@ -83,26 +86,39 @@ This repository provides blocks that can help you create a Dropdown, Select or a
 | `'visible'` | Displays a backdrop overlay which blocks the click on other items of the page, if you click on the overlay it will close the popover |
 | `'none'` | Doesn't display any backdrop overlay and if you click any item in the page it will close the popover and click the element |
 
+### Offsets
+
+| Property | Type | Description |
+| --- | --- | --- |
+| `skidding` | `number` | Displaces the `popover-layout` along the `popover-trigger`. |
+| `distance` | `number` | Displaces the `popover-layout` away from, or toward, the `popover-trigger` in the direction of its placement. A positive number displaces it further away, while a negative number lets it overlap the `popover-trigger`. |
```
Comment (klzns) [inline docs/README.md:94]:
```suggestion
| Value | Description |
| --- |  --- |
| `skidding` | Displaces the `popover-layout` along the `popover-trigger`. |
| `distance` | Displaces the `popover-layout` away from, or toward, the `popover-trigger` in the direction of its placement. A positive number displaces it further away, while a negative number lets it overlap the `popover-trigger`. |
```

File: docs/README.md [hunk 8]
```diff
@@ -83,26 +86,39 @@ This repository provides blocks that can help you create a Dropdown, Select or a
...
+
+
 ### PopoverTrigger
 
 | Prop name | Type | Description | Default value |
 | --- | --- | --- | --- |
-| trigger | `TriggerMode` | What kind of user action must happen to open the popover | `'click'` |
+| trigger | `TriggerMode` | What kind of user action must happen to open the `popover-layout` | `'click'` |
```
Comment (klzns) [inline docs/README.md:101]:
```suggestion
| trigger | `TriggerMode` | What kind of user action must happen to open the `popover-layout`.<br/> - `click`<br/>- `hover` | `'click'` |
```

File: docs/README.md [hunk 9]
```diff
@@ -83,26 +86,39 @@ This repository provides blocks that can help you create a Dropdown, Select or a
...
 
 ### TriggerMode
+
+| Value | Description |
 | --- | --- |
-| `'click'` | Open when you click the trigger |
-| `'hover'` | Open when you hover the trigger | 
+| `'click'` | Open when you click the `popover-trigger` |
+| `'hover'` | Open when you hover the `popover-trigger` |
```
Comment (klzns) [inline docs/README.md]: Remove

File: docs/README.md [hunk 10]
```diff
@@ -83,26 +86,39 @@ This repository provides blocks that can help you create a Dropdown, Select or a
...
 
+### Offsets
+
+| Property | Type | Description |
+| --- | --- | --- |
+| `skidding` | `number` | Displaces the `popover-layout` along the `popover-trigger`. |
+| `distance` | `number` | Displaces the `popover-layout` away from, or toward, the `popover-trigger` in the direction of its placement. A positive number displaces it further away, while a negative number lets it overlap the `popover-trigger`. |
+
+
 ### PopoverTrigger
```
Comment (klzns) [inline docs/README.md]:
```suggestion
## PopoverTrigger
```

Comment (klzns) [reply docs/README.md]: Do the same thing to ### Popover

**Review by klzns — CHANGES_REQUESTED** [review]

File: docs/README.md (hunk 8 above)
Comment (Klynger) [reply docs/README.md:101]: I think we could talk with @guerreirobeatriz about how to do this and change the structure of the documentation in general

**Review by guerreirobeatriz — APPROVED** [review]

**Review by klzns — APPROVED** [review]

**Review by victorhmp — APPROVED** [review]

**Pull request merged**